src/rust/dbv2/migrations/2024-01-09-160715_price_to_nominal/up.sql

```diff
@@ -0,0 +1,35 @@
+-- Your SQL goes here
+CREATE FUNCTION integer_price_to_quote_indivisible_subunits(market_id numeric, price numeric) RETURNS NUMERIC IMMUTABLE AS $$
+    SELECT ($2 / ((tick_size / POW(10::numeric,quote.decimals)) / (lot_size / POW(10,COALESCE(base.decimals, 0)))))
```

Comment:
I think the first division (`$2 /`) should actually be a multiplication, and also that the parens are off (there might be more issues)

This could probably be fixed by consolidating the algebra

How about?

```sql
SELECT ($2 * tick_size * POW(10::numeric, COALESCE(base.decimals, 0) - quote.decimals) / lot_size)
```

Reply:
How about `SELECT ($2 * tick_size * POW(10::numeric, COALESCE(base.decimals, 0))) / lot_size` so that it's in quote indivisible subunits ?

Reply:
This took me embarrassingly too much time to get right.

Reply:
Not sure on the intermediate, but the end result checks out